src/client/components/builder/ConstantsTab.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { FC, useEffect, useState } from 'react'
+import { BiPlusCircle, BiUpArrowAlt, BiDownArrowAlt } from 'react-icons/bi'
+import {
+  Center,
+  Heading,
+  Text,
+  VStack,
+  Button,
+  Menu,
+  MenuButton,
+} from '@chakra-ui/react'
+
+import * as checker from '../../../types/checker'
+import { FloatingToolbar } from './FloatingToolbar'
+import { MapTable } from './constants'
+
+import { useCheckerContext } from '../../contexts'
+
+import { BuilderActionEnum, ConfigArrayEnum } from '../../../util/enums'
+
+const generateDefaultMap = (id: number): checker.Constant => ({
+  id: `T${id}`,
+  title: 'Insert map title',
+  table: [{ key: '', value: NaN }],
+})
+
+export const ConstantsTab: FC = () => {
+  const [activeIndex, setActiveIndex] = useState<number>(0)
+  const [offsetTop, setOffsetTop] = useState<number>(16)
+  const [nextUniqueId, setNextUniqueId] = useState<number>(1)
+  const { config, dispatch } = useCheckerContext()
+
+  const { constants } = config
+
+  useEffect(() => {
+    let highestIndex = 0
+    config.constants.forEach((constant) => {
+      const constantIndex = parseInt(constant.id.slice(1))
+      highestIndex = Math.max(highestIndex, constantIndex)
+    })
+    setNextUniqueId(highestIndex + 1)
+  }, [config])
+
+  const toolbarOptions = [
+    {
+      icon: <BiPlusCircle />,
+      label: 'Add map',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Add,
+          payload: {
+            element: generateDefaultMap(nextUniqueId),
+            configArrName: ConfigArrayEnum.Constants,
+            newIndex: activeIndex + 1,
+          },
+        })
+        setActiveIndex(activeIndex + 1)
+        setNextUniqueId(nextUniqueId + 1)
+      },
+    },
+    {
+      icon: <BiUpArrowAlt />,
+      label: 'Move up',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Reorder,
+          payload: {
+            currIndex: activeIndex,
+            newIndex: activeIndex - 1,
+            configArrName: ConfigArrayEnum.Constants,
+          },
+        })
+        setActiveIndex(activeIndex - 1)
+      },
+      disabled: activeIndex === 0,
+    },
+    {
+      icon: <BiDownArrowAlt />,
+      label: 'Move down',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Reorder,
+          payload: {
+            currIndex: activeIndex,
+            newIndex: activeIndex + 1,
+            configArrName: ConfigArrayEnum.Constants,
+          },
+        })
+        setActiveIndex(activeIndex + 1)
+      },
+      disabled: activeIndex === constants.length - 1,
+    },
+  ]
+
+  const onSelect = ({ index }: { index: number }) => {
+    setActiveIndex(index)
+  }
+
+  const onActive = ({ top }: { top: number }) => {
+    setOffsetTop(top)
+  }
+
+  const addInitialMap = () => {
+    dispatch({
+      type: BuilderActionEnum.Add,
+      payload: {
+        element: generateDefaultMap(0),
```

Comment:
```suggestion
        element: generateDefaultMap(nextUniqueId),
```

Currently, the first constant has an ID of `T0`. We should probably standardise the numbering to start from 1, similar to questions and logic.

Reply:
addressed with cb2795f